### PR TITLE
[Excel Formatting] Add feature flag to enable export data typing on form and case exports

### PR DESCRIPTION
**Pull request by biyeun**
https://dimagi-dev.atlassian.net/browse/SAASP-10249

##### SUMMARY
- Added a checkbox "Automatically format cells for Excel 2007+" to enable the cell data typing functionality ONLY on that particular form or case export that has that checkbox ticked.
- Added support for ISO-formatted date strings (raw dates from Couch) which could be part of exports when `transform_dates` ("Automatically convert dates and multimedia links for Excel") is unchecked.
- Put all of the above under a feature flag so that high priority clients can get access ASAP on production. I will be putting this on staging for a quick smoke test, but the main QA process still needs to happen to test this as a formal product feature.

##### FEATURE FLAG
`EXCEL_EXPORT_DATA_TYPING` -> Enable the "Automatically format cells for Excel 2007+" checkbox in form and case exports.

##### PRODUCT DESCRIPTION
Adds a new checkbox in the UI (disabled/false by default) to enable cell formatting in excel 2007+ exports. It also updates the text on the `transform_dates` checkbox to make it a little clearer what that checkbox does.

Screen Shot 2020-01-21 at 4 04 22 PM

### CODE REVIEW

File: corehq/ex-submodules/couchexport/util.py [hunk 1]
```diff
@@ -200,10 +200,10 @@ def get_excel_format_value(value):
         return ExcelFormatValue(numbers.FORMAT_NUMBER_00, value)
     if isinstance(value, Decimal):
         return ExcelFormatValue(numbers.FORMAT_NUMBER_00, float(value))
-    if isinstance(value, datetime.date):
-        return ExcelFormatValue(numbers.FORMAT_DATE_YYYYMMDD2, value)
     if isinstance(value, datetime.datetime):
         return ExcelFormatValue(numbers.FORMAT_DATE_DATETIME, value)
+    if isinstance(value, datetime.date):
+        return ExcelFormatValue(numbers.FORMAT_DATE_YYYYMMDD2, value)
```
Comment (dannyroberts) [inline corehq/ex-submodules/couchexport/util.py]: Ugh classic. It's been a while but I've hit this bizarre aspect of the date / datetime class hierarchy before

File: corehq/apps/export/templates/export/customize_export_new.html [hunk 2]
```diff
@@ -111,17 +111,42 @@
...
+                  {% endif %}
                 </label>
               </div>
 
+              {% if request|toggle_enabled:"EXCEL_EXPORT_DATA_TYPING" %}
+              <div class="checkbox">
+                <label>
+                  <input type="checkbox"
+                         id="format-data-in-excel-checkbox"
+                         data-bind="checked: format_data_in_excel" />
```
Comment (dannyroberts) [inline corehq/apps/export/templates/export/customize_export_new.html]: This is minor UX feedback probably for after this is merged and when we're releasing the feature more widely, but this only affects Excel 2007+, right? Maybe it'd make sense to grey/out & disable the checkbox unless Excel 2007+ format is selected?

**Review by dannyroberts — APPROVED** [review]

**Comment by dannyroberts** [comment]
Looks great to me!

File: corehq/apps/export/templates/export/customize_export_new.html (hunk 2 above)
Comment (biyeun) [reply corehq/apps/export/templates/export/customize_export_new.html]: yeah I definitely thought about that. it was a little more of an issue due to how the knockout code for exports is structured. it wasn't super straightforward to add in that logic and would require some digging. If we decide it's worth the extra time, I'm happy to do it.

**Pull request merged**